### DISCUSSION TITLE
Joins/Select-IN-Loads on Row-Limited Relationships with Window Functions

### DISCUSSION THREAD

**Discussion by pythonweb2** [sqlalchemy/sqlalchemy#9903] [Usage Questions] [answered]
In the docs, there is a section on [Row-Limited Relationships with Window Functions](https://docs.sqlalchemy.org/en/20/orm/join_conditions.html#row-limited-relationships-with-window-functions).

One of the downsides to this approach seems to be that no options can be used on the relationship that is created. For example, if we take the example code and try to use it like this:

```python
session.scalars(select(A).options(selectinload(A.partitioned_bs).joinedload(B.c))
```
(Assuming there is a table called C and a relationship on B that links to it).

We get an error like this:
```
sqlalchemy.exc.ArgumentError: ORM mapped entity or attribute "B.c" does not link from relationship "A.partitioned_bs.of_type(aliased(B))".
```

Is there a way to get around this? Or is this just not a supported feature at this time?

**Comment by zzzeek** [comment]
I didn't think it had that limitation in this case, but try:

     selectinload(A.partitioned_bs.of_type(B)).joinedload(B.c)

**Reply by zzzeek** [reply, marked as answer]
or as mentioned in [the docs](https://docs.sqlalchemy.org/en/20/orm/join_conditions.html#using-the-aliasedclass-target-in-queries), join to that aliased class:

    selectinload(A.partitioned_bs).joinedload(partitioned_b.c)

same thing

**Reply by pythonweb2** [reply]
Oh whoops didn't see that in the docs. 

First one didn't work, but the second one did the trick, thank you!
